drivers/video/gc2145.c

```diff
@@ -808,87 +746,49 @@ static int gc2145_soft_reset(const struct device *dev)
 
 static int gc2145_set_ctrl_vflip(const struct device *dev, bool enable)
 {
-	int ret;
 	const struct gc2145_config *cfg = dev->config;
-	uint8_t old_value;
-
-	ret = gc2145_read_reg(&cfg->i2c, GC2145_REG_AMODE1, &old_value);
-	if (ret < 0) {
-		return ret;
-	}
 
 	/* Set the vertical flip state */
-	return gc2145_write_reg(&cfg->i2c, GC2145_REG_AMODE1,
-				(old_value & GC2145_AMODE1_WINDOW_MASK) | (enable << 1));
+	return video_modify_cci_reg(&cfg->i2c, GC2145_REG_AMODE1, GC2145_AMODE1_WINDOW_MASK,
+				    FIELD_PREP(GC2145_AMODE1_WINDOW_MASK, enable << 1));
```

Comment:
If using `field_prep`, I must remove the `<< 1`.

The HFLIP/VFLIP was not tested as the UVC standard does not support HFLIP/VFLIP.
This involves a number of challenges as this can altern the pixel format like reversing two color channels, and GC2145 is even affected by this.

Solutions to test it anyway:

- Map all V4L2 controls to UVC extended controls:
  https://github.com/zephyrproject-rtos/zephyr/issues/86943
- Use the Zephyr Video Shell:
  https://github.com/zephyrproject-rtos/zephyr/pull/88566 (rebase needed and then can be merged)